### PR TITLE
Bug/skaled 1623 sigterm at exit

### LINKED ISSUE #1623: SIGETRM during shutdown forces shutdown
When skaled is shutting down by internal cause, it counts this event as though it was external SIGTERM. This leads to forced shutdown if real SIGTERM is received during shutdown procedure. Need not to count internal exit request as SIGTERM.

### PR DESCRIPTION
**Pull request by dimalit**
When skaled is shutting down by internal cause, it counts this event as though it was external SIGTERM. This leads to forced shutdown if real SIGTERM is received during shutdown procedure. Need not to count internal exit request as SIGTERM.

Test https://github.com/skalenetwork/skale-node-tests/blob/master/test_stop.py#L252

### CODE REVIEW

**Comment by dimalit** [comment]
Probably, let's wait for "stable" - it will integrate changes in develop into v3.18.0 - and this PR would not be so bloated

File: libdevcore/Common.cpp [hunk 1]
```diff
@@ -47,11 +47,18 @@ void ExitHandler::exitHandler( int s ) {
...
 void ExitHandler::exitHandler( int nSignalNo, ExitHandler::exit_code_t ec ) {
-    std::string strMessagePrefix = ExitHandler::shouldExit() ?
-                                       cc::error( "\nStop flag was already raised on. " ) +
-                                           cc::fatal( "WILL FORCE TERMINATE." ) +
-                                           cc::error( " Caught (second) signal. " ) :
-                                       cc::error( "\nCaught (first) signal. " );
+    std::string strMessagePrefix;
+    if ( nSignalNo > 0 ) {
+        strMessagePrefix = ( ExitHandler::shouldExit() && s_nStopSignal > 0 ) ?
+                               cc::error( "\nStop flag was already raised on. " ) +
```
Comment (kladkogex) [inline libdevcore/Common.cpp]: Please remove color comments

File: libdevcore/Common.cpp [hunk 2]
```diff
@@ -47,11 +47,18 @@ void ExitHandler::exitHandler( int s ) {
...
+                                   cc::error( " Caught (second) signal. " ) :
+                               cc::error( "\nCaught (first) signal. " );
+    } else {
+        strMessagePrefix = ExitHandler::shouldExit() ?
+                               cc::error( "\nInternal exit requested while already exiting. " ) :
+                               cc::error( "\nInternal exit initiated. " );
+    }
     std::cerr << strMessagePrefix << cc::error( skutils::signal::signal2str( nSignalNo ) )
               << "\n\n";
     std::cerr.flush();
```
Comment (kladkogex) [inline libdevcore/Common.cpp]: Remove this line since cerr is always flushed on line end

File: libdevcore/Common.cpp [hunk 3]
```diff
@@ -153,7 +160,8 @@ void ExitHandler::exitHandler( int nSignalNo, ExitHandler::exit_code_t ec ) {
 
     // nice exit here:
 
-    if ( ExitHandler::shouldExit() ) {
+    // TODO deduplicate with first if()
+    if ( ExitHandler::shouldExit() && s_nStopSignal > 0 && nSignalNo > 0 ) {
         std::cerr << ( "\n" + cc::fatal( "SIGNAL-HANDLER:" ) + " " +
                        cc::error( "Will force exit now..." ) + "\n\n" );
         _exit( 13 );
```
Comment (kladkogex) [inline libdevcore/Common.cpp:162]: Why is 13 used here ? Is there a document in which error codes are defined?

**Review by kladkogex — CHANGES_REQUESTED** [review]

**Review by DmytroNazarenko — DISMISSED** [review]
Approved, next steps:
1. Need to add test to functional-tests
2. Fix colorization
4. Test build on devnet

**Comment by dimalit** [comment]
UPD Currently, functional tests fail in load_js. But fail happens because of skaled JSON-RPC request refusal. As changes in this PR cannot affect JSON-RPC at all - most probably, issue is because of Debug build instead of Release build. It should be gone after merge and Release build.

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**